### PR TITLE
feat(screen-lock): request on conference join.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: react/features/base/conference/middleware.web.ts [hunk 1]
```diff
@@ -23,6 +42,13 @@ MiddlewareRegistry.register(store => next => action => {
             dispatch(setSkipPrejoinOnReload(false));
         }
 
+        if (navigator.wakeLock?.request) {
+            navigator.wakeLock.request('screen')
+                .then(lock => {
```
Comment (saghul) [inline react/features/base/conference/middleware.web.ts:47]: Maybe add a catch with a log?

File: react/features/base/conference/middleware.web.ts [hunk 2]
```diff
@@ -32,8 +58,15 @@ MiddlewareRegistry.register(store => next => action => {
             dispatch(setSkipPrejoinOnReload(true));
         }
 
+        releaseScreenLock();
+
         break;
     }
+    case CONFERENCE_LEFT:
+    case KICKED_OUT:
+        releaseScreenLock();
```
[the remark this answers is not in the document]
Comment (hristoterezov) [reply react/features/base/conference/middleware.web.ts:67]: It seems to me that kick is not touching the conference. I'd prefer to leave it as it is.

Shouldn't the order for the breakout room be leave and then join? wouldn't we mess up the conference object in redux if this is not the order?

[the remark this answers is not in the document]
Comment (saghul) [reply react/features/base/conference/middleware.web.ts:71]: I'd like us to think about consistency. Web is more forgiving because the page is reloaded with every meeting, not so on mobile. The state listener patter is pretty wide spread across the code base at this point.

[the remark this answers is not in the document]
Comment (hristoterezov) [reply react/features/base/conference/middleware.web.ts:71]: - I don't think there's an issue with consistency because middlewares on those actions are widely used.
 - The feature is specific to web
 - The breakout rooms events will arrive in the correct order see - https://github.com/jitsi/jitsi-meet/blob/6a21a7e74ba35374c3834a479454ca315973a0ad/react/features/breakout-rooms/actions.ts#L229
 - In order to implement it the way you are asking I need to start clearing the conference state on kick. Let's not over-engineer this otherwise simple PR!

File: tsconfig.web.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
         "module": "es2020",
         "target": "es2020",
         "jsx": "react",
-        "lib": [ "webworker", "ES2020", "DOM" ],
+        "lib": [ "ES2020", "DOM" ],
```
Comment (saghul) [inline tsconfig.web.json:8]: Why did you need to remove this? No adverse side effects?

**Review by saghul — APPROVED** [review]
LGTM with a question.

File: tsconfig.web.json (hunk 3 above)
Comment (hristoterezov) [reply tsconfig.web.json:8]: It turns out when we add "webworker" in lib TS is treating everything as workers. The result is that here I use types for the navigator.wakeLock which are adding the wakeLock prop to the Navigator interface but we get an error when "webworker" lib is added in the config because the type of navigator is WorkerNavigator. It seems to me that if we need the webworker lib option we will need to add some extra care about making TS to use the option only for the workers somehow.

It doesn't seems to have any adverse effects with the existing code base. Maybe now the workers won't be so restricted by TS but I haven't tested it. Anyway treating all files as workers doesn't sound correct too.
